src/vs/workbench/services/textfile/common/textFileSaveParticipant.ts

```diff
@@ -32,25 +36,41 @@ export class TextFileSaveParticipant extends Disposable {
 		// undoStop before participation
 		model.textEditorModel?.pushStackElement();
 
-		for (const saveParticipant of this.saveParticipants) {
-			if (token.isCancellationRequested || !model.textEditorModel /* disposed */) {
-				break;
-			}
+		const cts = new CancellationTokenSource(token);
+
+		await this.progressService.withProgress({
+			priority: NotificationPriority.URGENT,
+			location: ProgressLocation.Notification,
```

Comment:
Are you intentionally picking `ProgressLocation.Notification` here to show it to the user? Because the overall save is reported on `ProgressLocation.Window` here: https://github.com/microsoft/vscode/blob/b7a9323c5213f3c1c104304c10645f54bd2ffc0c/src/vs/workbench/services/textfile/common/textFileEditorModel.ts#L825

Reply:
Yeah, I did that on purpose so that it's really clear what's blocking save. Tho, I don't know if calling out the extensions is fair because the whole operation can be blocked 

* because of a slow/bad save participant, 
* or because of unresponsive EH

In the latter case we will "blame" the save participant extension which are equally "stuck in traffic" as anything else.